### PR TITLE
perp-4102 | burn half of the house LP shares

### PR DESCRIPTION
**Pull request by lvn-rusty-dragon**
This is an incentivization mechanism to provide additional value over time to liquidity providers.

### CODE REVIEW

File: contract/src/execute.rs [hunk 1]
```diff
@@ -338,7 +338,8 @@ fn withdraw(
 
     let Sell { funds, returned } = market.sell(outcome, tokens)?;
 
-    if share_info.get_outcome(&market, outcome)?.is_zero() {
+    if share_info.get_outcome(&market, outcome, false)?.is_zero() {
+        println!("here1");
```
Comment (lvn-hasky-dragon) [inline contract/src/execute.rs:342]: I believe you didn't mean to check this in..

File: contract/src/sanity.rs [hunk 2]
```diff
@@ -83,6 +83,7 @@ pub fn sanity(store: &dyn Storage, env: &Env) {
             assert_eq!(tokens.0, pool_size.0);
         }
 
+        // FIXME modify check to include in all wallets if they have a liquidity share
```
Comment (lvn-hasky-dragon) [inline contract/src/sanity.rs:86]: Not sure if you want to address this as part of this PR or later.

**Review by lvn-hasky-dragon — APPROVED** [review]

File: contract/src/sanity.rs (hunk 2 above)
Comment (lvn-rusty-dragon) [reply contract/src/sanity.rs:86]: Fixed in the next PR.

File: contract/src/execute.rs (hunk 1 above)
Comment (lvn-rusty-dragon) [reply contract/src/execute.rs]: Doh, good catch, thanks!

**Pull request merged**